src/Effect/Exec.hs

```diff
@@ -287,6 +285,238 @@ instance ToDiagnostic ExecErr where
+--
+-- The function searches for the binary with the first found of @Extensions@,
+-- in the first of provided @Directories@.
+-- The search is conducted in the order of directories -> extensions.
+--
+-- If multiple binaries are intended, see @which'@.
+--
+-- If all directories and extensions are exhausted without finding a match,
+-- the function returns @Nothing@.
+--
+-- * @Directories@
+-- The function begins its search in the provided working directory and all of its parent directories.
+-- After that, it searches all entries in the system @$PATH@ environment variable.
+--
+-- * @Extensions@
+-- In Unix-based systems, @Extensions@ is always an empty list.
+-- In Windows-based systems, the function considers all entries in the system's
+-- @$PATHEXT@ environment variable.
+--
+-- * Examples:
+--
+-- __Windows__
+--
+-- > Binary : "mvn"
+-- > PATH   : "C:\System32"
+-- > PATHEXT: ".bat;.exe"
+-- > WORKDIR: "C:\Users\me\projects\example"
+--
+-- Searches:
+--
+-- > C:\Users\me\projects\example\mvn.exe
+-- > C:\Users\me\projects\example\mvn.bat
+-- > C:\Users\me\projects\mvn.exe
+-- > C:\Users\me\projects\mvn.bat
+-- > C:\Users\me\mvn.exe
+-- > C:\Users\me\mvn.bat
+-- > C:\Users\mvn.exe
+-- > C:\Users\mvn.bat
+-- > C:\mvn.exe
+-- > C:\mvn.bat
+-- > C:\System32\mvn.exe
+-- > C:\System32\mvn.bat
+--
+-- __Linux, macOS__
+--
+-- > Binary : "mvn"
+-- > PATH   : "/usr/local/bin"
+-- > WORKDIR: "/home/me/projects/example"
+--
+-- Searches:
+--
+-- > /home/me/projects/example/mvn
+-- > /home/me/projects/mvn
+-- > /home/me/mvn
+-- > /home/mvn
+-- > /mvn
+-- > /usr/local/bin/mvn
+which :: (WhichEffs sig m) => Path Abs Dir -> Text -> m (Maybe FoundLocation)
+which workdir bin = which' workdir (singleton bin)
+
+-- | Find the full path to an executable on the system.
+--
+-- The function searches for the first found in the list of @Binaries@,
+-- with the first found of @Extensions@, in the first of provided @Directories@.
+-- The search is conducted in the order of directories -> binaries -> extensions.
+--
+-- If all directories, binaries, and extensions are exhausted without finding a match,
+-- the function returns @Nothing@.
+--
+-- * @Directories@
+-- The function begins its search in the provided working directory and all of its parent directories.
+-- After that, it searches all entries in the system @$PATH@ environment variable.
+--
+-- * @Extensions@
+-- In Unix-based systems, @Extensions@ is always an empty list.
+-- In Windows-based systems, the function considers all entries in the system's
+-- @$PATHEXT@ environment variable.
+--
+-- * @Binaries@
+-- The function can be provided with multiple binary names.
+-- This is useful in cases where a binary might have different names,
+-- but the usage of the binary is exactly the same regardless of the name used.
+-- The binary names are searched in order, and the function returns the path of the first one found.
+--
+-- * Examples:
+--
+-- __Windows__
+--
+-- > Binaries: [ "mvn", "mvnw" ]
```

Comment:
If, in practice, mvnw is preferred over mvn, then I think this example should reflect this

Reply:
So uh, it turns out, that I had messed up the actual order 👀 
But it's [fixed now in the latest commit](https://github.com/fossas/fossa-cli/pull/1209/commits/e187e49071c84254eb4cdea45f0568e93b4d65a5#diff-44189d9cbe979286ff7d5111b3e7f795518942972009ea8f436f855c8438b82e)...

Reply:
Glad you noticed this!